File: updraft_sdk/src/main/java/com/apsswithlove/updraft_sdk/feedback/form/FeedbackFormContract.java
package com.apsswithlove.updraft_sdk.feedback.form;

public interface FeedbackFormContract {

    interface View {

        FeedbackChoice getSelectedChoice();

        String getEmail();

        String getDescription();

        void showProgress();

        void hideProgress();

        void showSuccessMessage();

        void showErrorMessage(Throwable t);

        void updateProgress(double progress);

        void closeFeedback();
    }
}

File: updraft-sdk/src/main/java/com/appswithlove/updraft/feedback/form/FeedbackFormContract.kt
package com.appswithlove.updraft.feedback.form

interface FeedbackFormContract {

    interface View {
        fun getSelectedChoice(): FeedbackChoice
        fun getEmail(): String
        fun getDescription(): String

        fun showProgress()
        fun hideProgress()
        fun showSuccessMessage()
        fun showErrorMessage(t: Throwable)
        fun updateProgress(progress: Double)
        fun closeFeedback()
    }
}
